submit/controllers/ui/new/process.py

```diff
@@ -199,7 +199,25 @@ def start_compilation(params: MultiDict, session: Session, submission_id: int,
     response_data.update(**result.extra)
 
     if result.status == process_source.FAILED:
-        alerts.flash_failure(f"Processing failed")
+        if 'reason' in result.extra and "produced from TeX source" in result.extra['reason']:
+            msg = Markup("The submission PDF file appears to have been produced by TeX. <p>This file has been rejected " \
```

Comment:
It might be good to break out these extended reasons into a new module (like reasons.py) at some point.

Reply:
Done. Created new reasons.py file for detailed error/warning explanations.